spec/services/exports/generate_applicants_csv_spec.rb

```diff
@@ -81,48 +91,141 @@
       end
 
-      it "generates one line for each applicant required" do
-        expect(subject.csv.scan(/(?=\n)/).count).to eq(applicants.count + 1)
+      context "it exports all the concerned applicants" do
+        it "generates one line for each applicant required" do
+          expect(csv.scan(/(?=\n)/).count).to eq(4) # one line per applicants + 1 line of headers
+        end
+
+        it "displays all the applicants" do
+          expect(csv).to include("Doe")
+          expect(csv).to include("Casubolo")
+          expect(csv).to include("Blanc")
+        end
       end
 
-      it "displays all the applicants" do
-        csv = subject.csv
-        expect(csv).to include("Doe")
-        expect(csv).to include("Casubolo")
-        expect(csv).to include("Blanc")
+      context "it displays the right attributes" do
+        let!(:applicants) { Applicant.where(id: [applicant1]) }
+
+        it "displays the applicants attributes" do
+          expect(csv.scan(/(?=\n)/).count).to eq(2) # we check there is only one applicant for this test
+          expect(csv).to include("madame")
+          expect(csv).to include("Doe")
+          expect(csv).to include("Jane")
+          expect(csv).to include("12345") # affiliation_number
+          expect(csv).to include("33333") # department_internal_id
+          expect(csv).to include("20 avenue de Ségur 75OO7 Paris")
+          expect(csv).to include("jane@doe.com")
+          expect(csv).to include("01 01 01 01 01")
+          expect(csv).to include("20/12/1977") # birth_date
+          expect(csv).to include("20/05/2022") # rights_opening_date
+          expect(csv).to include("demandeur") # role
+        end
+
+        it "displays the invitations infos" do
+          expect(csv).to include("21/05/2022") # first invitation date
+          expect(csv).to include("22/05/2022") # last invitation date
+          expect(csv).not_to include("(Délai dépassé)") # invitation delay
+        end
+
+        it "displays the rdvs infos" do
+          expect(csv).to include("25/05/2022") # last rdv date
+          expect(csv).to include("25/05/2022;oui") # last rdv taken in autonomy ?
+          expect(csv).to include("Statut du RDV à préciser") # rdv_context status
+          expect(csv).to include("Statut du RDV à préciser;oui") # first rdv in less than 30 days ?
+          expect(csv).to include("25/05/2022;oui;Statut du RDV à préciser;oui;25/05/2022") # orientation date
+        end
+
+        context "when the invitation deadline has passed" do
+          let!(:rdv_context) do
+            create(
+              :rdv_context, rdvs: [], invitations: [first_invitation, last_invitation],
+                            applicant: applicant1, status: "invitation_pending"
+            )
+          end
+
+          it "displays 'délai dépassé'" do
+            expect(subject.csv).to include("Invitation en attente de réponse (Délai dépassé)") # rdv_context status
+          end
+        end
```

Comment:
On pourrait peut-être mettre ce blocs `context` après tous les blocs `it` ci-dessous